### PR TITLE
Move openhpc compute, runtime and control to its own repository

### PR DESCRIPTION
**Pull request by brtknr**
Conflicts:
	ansible/euclid-openhpc.yml
	ansible/requirements.yml
	ansible/roles/openhpc_runtime/handlers/main.yml
	ansible/roles/openhpc_runtime/tasks/main.yml
	ansible/roles/openhpc_runtime/tasks/runtime.yml
	ansible/roles/openhpc_runtime/tasks/slurm.yml
	ansible/roles/openhpc_runtime/templates/slurm.conf.j2

### CODE REVIEW

File: ansible/requirements.yml [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+- src: singleplatform-eng.users
+- src: stackhpc.beegfs
 - src: stackhpc.cluster-infra
-- src: stackhpc.monasca-agent
-- src: stackhpc.monasca-rsyslog
 - src: stackhpc.grafana-conf
 - src: stackhpc.gluster-cluster
+- src: stackhpc.monasca-agent
+- src: stackhpc.monasca-rsyslog
+- src: stackhpc.openhpc
```
Comment (markgoddard) [inline ansible/requirements.yml]: How about a version?

File: ansible/slurm.yml [hunk 2]
```diff
@@ -1,21 +1,12 @@
...
-- hosts: cluster_batch
-  become: yes
-  roles:
-    - role: openhpc_compute
-
-# Cluster setup
-- hosts: 
-  - cluster_login
+- hosts:
+  - cluster_control
```
Comment (oneswig) [inline ansible/slurm.yml]: We probably want to include cluster_login here as well, to ensure it gets the OpenHPC runtime environment installed.

**Comment by markgoddard** [comment]
Has a merge conflict.

File: ansible/slurm.yml (hunk 2 above)
Comment (markgoddard) [reply ansible/slurm.yml]: +1

**Comment by brtknr** [comment]
All issues resolved!

File: ansible/slurm-drain.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
+- import_playbook: cluster-infra-query.yml
```
Comment (oneswig) [inline ansible/slurm-drain.yml]: Is it worthwhile making a separate playbook for cluster-infra-query?  Might be equally succinct to invoke `stackhpc.cluster-infra` directly here and save a layer.

File: ansible/slurm-drain.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+- hosts: openstack
+  gather_facts: false
+  tasks:
+    - name: Count the number of compute nodes per slurm partition
+      set_fact:
+        desired_state: "{{ ((cluster_group.output_value | selectattr('group', 'equalto', item.name ) | first).nodes | map(attribute='name') | list )[:item.num_nodes] + desired_state | default([]) }}"
```
Comment (oneswig) [inline ansible/slurm-drain.yml]: Does this work if an expected partition is not found in the running stack?  Might be worth checking for that first if not.  Similarly, if an extra partition is unexpectedly found

File: ansible/slurm-drain.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+- hosts:
+  - cluster_batch
+  become: yes
+  roles:
+    - role: stackhpc.openhpc
+      openhpc_cluster_name: "{{ cluster_name }}"
+      openhpc_slurm_control_host: "{{ groups['cluster_control'] | first }}"
+      openhpc_enable:
+        control: false
```
Comment (oneswig) [inline ansible/slurm-drain.yml]: These new parameters are a little fishy - we are kind of implying a deployment without anything in it, which is a counterintuitive way of having the effect we want (which is not to deploy anything).  Thankfully, *ahem* this mode of operation's going to be clearly documented in the role's README...

File: ansible/slurm-drain.yml (hunk 3 above)
Comment (brtknr) [reply ansible/slurm-drain.yml]: Makes sense

File: ansible/slurm-drain.yml (hunk 4 above)
Comment (brtknr) [reply ansible/slurm-drain.yml]: It did fail but I have address this in a new commit.

File: ansible/slurm-drain.yml (hunk 5 above)
Comment (brtknr) [reply ansible/slurm-drain.yml]: I'm not 100% clear about what you mean here.

**Comment by brtknr** [comment]
I'll merge this after the PR in openhpc role is merged:

**Pull request merged**